### DISCUSSION TITLE
Field calculations using Functional

### DISCUSSION THREAD

**Discussion by vr0110** [kinnala/scikit-fem#968] [Q&A] [answered]
Hi everybody.
This must be an easy question to answer, but what is the right way to perform calculations on a (scalar) solution $f$, such as $df/dy$?

After many failed attempts, I tried the following minimal code (based on [example 1](https://scikit-fem.readthedocs.io/en/latest/listofexamples.html#example-1-poisson-equation-with-unit-load)). Before doing any calculations, I just tried to return the field without any modifications. The results `x` and `x2` are not the same!

It would be nice if someone could provide any hints or explanations. Thanks in advance!

MWE:
```python
import skfem
from skfem import *
from skfem.models import laplace, unit_load
from skfem.visuals.matplotlib import plot, show

m = MeshTri().refined(6)
e = ElementTriP1()
basis = Basis(m, e)
A = laplace.assemble(basis)
b = unit_load.assemble(basis)
A, b = enforce(A, b, D=m.boundary_nodes())
x = solve(A, b)


@skfem.Functional
def f(w):
    field = w['field']
    return field


x2 = f.elemental(basis, field=x)

plot(m, x, colorbar=True)
plot(m, x2, colorbar=True)
show()

```

**Comment by vr0110** [comment]
Here is the output. It looks similar, but the scale is off:
![x](https://user-images.githubusercontent.com/117990222/201339914-dfc4ab91-9504-48b5-be75-35cf999d800d.png)
![x2](https://user-images.githubusercontent.com/117990222/201339930-ab619cc1-923a-47c1-a887-efda8bbed2c9.png)

**Reply by kinnala** [reply]
Functional is basically an integral over the domain. If you use f.elemental it calculates the integral element-by-element. So the latter is basically the former multiplied the square of the element size.

**Comment by vr0110** [comment, marked as answer]
Ok, then `Functional` is the wrong way. I don't want to integrate anything.
So, how to calculate things like the derivative on the solution? More specifically, I need to calculate $(df_1/dx)^2+(df_2/dx)^2$ with $f_1$ and $f_2$ being two individual solutions (on the same mesh and same element).

**Reply by kinnala** [reply]
There are different ways but perhaps the simplest is basis.project(basis.interpolate(x).grad[0]).

**Comment by vr0110** [comment]
Oh, that's it. I actually read the [howto on projections](https://scikit-fem.readthedocs.io/en/latest/howto.html#performing-projections) many times, but just did not see how to apply it to my problem. Thank you very much!

Also thank you for this awesome awesome piece of open source software!
